cmake/global_functions.cmake

```diff
@@ -0,0 +1,266 @@
+            list(APPEND CHECK_PATH_LIST "${MISSION_DEFS}/${PREFIX}_${GENCONFIG_ARG_MATCH_SUFFIX}")
+        endforeach()
+    endif(GENCONFIG_ARG_MATCH_SUFFIX)
+
+    # Check for existence of files, and add to content if present
+    # Note that all files are appended/concatenated together.
+    foreach(SRC_LOCAL_PATH ${CHECK_PATH_LIST})
+        if (EXISTS "${SRC_LOCAL_PATH}")
+            file(TO_NATIVE_PATH "${SRC_LOCAL_PATH}" SRC_NATIVE_PATH)
+            list(APPEND WRAPPER_FILE_CONTENT "#include \"${SRC_NATIVE_PATH}\"\n")
+            set(ITEM_FOUND TRUE)
+        else()
+            list(APPEND WRAPPER_FILE_CONTENT "/* ${SRC_LOCAL_PATH} does not exist */\n")
+        endif (EXISTS "${SRC_LOCAL_PATH}")
+    endforeach()
+
+    # If _no_ files were found in the above loop,
+    # then check for and use the fallback file.
+    # (if specified by the caller it should always exist)
+    # Also produce a message on the console showing whether mission config or fallback was used
+    if (ITEM_FOUND)
+        message(STATUS "Generated ${GENCONFIG_ARG_FILE_NAME} from ${MISSION_DEFS} configuration")
+    elseif (GENCONFIG_ARG_FALLBACK_FILE)
+        file(TO_NATIVE_PATH "${GENCONFIG_ARG_FALLBACK_FILE}" SRC_NATIVE_PATH)
+        list(APPEND WRAPPER_FILE_CONTENT
+            "\n\n/* No configuration for ${GENCONFIG_ARG_FILE_NAME}, using fallback */\n"
+            "#include \"${GENCONFIG_ARG_FALLBACK_FILE}\"\n")
+        message(STATUS "Using ${GENCONFIG_ARG_FALLBACK_FILE} for ${GENCONFIG_ARG_FILE_NAME}")
+    else()
+        message("ERROR: No implementation for ${GENCONFIG_ARG_FILE_NAME} found")
+        message(FATAL_ERROR "Tested: ${CHECK_PATH_LIST}")
+    endif()
+
+    # Generate a header file
+    generate_c_headerfile("${GENCONFIG_ARG_OUTPUT_DIRECTORY}/${GENCONFIG_ARG_FILE_NAME}" ${WRAPPER_FILE_CONTENT})
+
+endfunction(generate_config_includefile)
+
+
+##################################################################
+#
+# FUNCTION: read_targetconfig
+#
+# Scan the list of targets and organize by target system type.
+# This function sets up the following variables in the global scope:
+#   TGTSYS_LIST: list of CPU architectures used in the build.  Note this
+#       will always contain a "native" target (for tools at least) which 
+#       is forced to be last.
+#   MISSION_APPS: full list of applications specified in the whole mission
+#   SYSID_<arch>: set for each entry of TGTSYS_LIST, and indicates the
+#       toolchain specified in the target file for that CPU arch.
+#   TGTSYS_<arch>: set to a list of CPU numbers that utilize the same arch 
+#   TGTSYS_<arch>_APPS: set for each entry of TGTSYS_LIST, and indicates the
+#       full set of applications that need to built for that target architecture
+#   TGTSYS_<arch>_DRIVERS: set for each entry of TGTSYS_LIST, and indicates the
+#       full set of device drivers that need to built for that target architecture
+#
```

Comment:
Documentation states that read_targetconfig creates variables TGTSYS_<arch>_DRIVERS and SYSID_<arch> which are not created in this function. Also might need to specify variables TGTSYS_<arch>_PSPMODULES, TGTSYS_<arch>_STATICAPPS

Reply:
See https://github.com/nasa/cFE/issues/1306